src/common/src/array/arrow/arrow_impl.rs

```diff
@@ -841,7 +907,43 @@ macro_rules! converts {
             }
         }
     };
+
+    ($ArrayType:ty, $ArrowType:ty, $time_unit:expr, @map) => {
+
+        impl From<&$ArrayType> for $ArrowType {
+            fn from(array: &$ArrayType) -> Self {
+                array.iter().map(|o| o.map(|v| v.into_arrow())).collect()
+            }
+        }
```

Comment:
It seems we need to have a method `into_arrow_with_unit()` as well.

Reply:
Here, the main consideration is compatibility. Previously, user sink timestamps with only a microsecond precision. If we change this to allow sinking in seconds, milliseconds, etc., downstream users might encounter multiple columns added. In the current implementation, zeros are only added when precision is insufficient, and there are no correctness issues. I suggest we maintain the current situation.